### PR TITLE
Htmx

### PR DESCRIPTION
**Pull request by amickan**
This PR adds HTMX to GC. A first use case is updating a challenge request status in place on the list view, and updating a request's budget fields in place on the detail view.

https://user-images.githubusercontent.com/30069334/165907906-bd37a582-2016-4e1b-93bc-7f372339aa56.mp4

### CODE REVIEW

File: app/grandchallenge/challenges/urls.py [hunk 1]
```diff
@@ -46,8 +47,13 @@
         name="requests-detail",
     ),
     path(
-        "requests/<pk>/update/",
-        ChallengeRequestUpdate.as_view(),
-        name="requests-update",
+        "requests/<pk>/update/status",
```
Comment (jmsmkn) [inline app/grandchallenge/challenges/urls.py:50]:
```suggestion
        "requests/<pk>/update/status/",
```

File: app/grandchallenge/challenges/urls.py [hunk 2]
```diff
@@ -46,8 +47,13 @@
...
+        ChallengeRequestStatusUpdate.as_view(),
+        name="requests-status-update",
+    ),
+    path(
+        "requests/<pk>/update/budget",
```
Comment (jmsmkn) [inline app/grandchallenge/challenges/urls.py:55]:
```suggestion
        "requests/<pk>/update/budget/",
```

File: app/tests/challenges_tests/test_forms.py [hunk 3]
```diff
@@ -78,3 +79,39 @@ def test_accept_challenge_request(
         f"There already is a challenge with short name: {type_1_challenge_request.short_name}"
         in str(form.errors)
     )
+
+
+@pytest.mark.django_db
+def test_budget_update_form(
+    client, challenge_reviewer, type_2_challenge_request
```
Comment (jmsmkn) [inline app/tests/challenges_tests/test_forms.py:86]: Are `client` and `challenge_reviewer` required here?

**Review by jmsmkn — COMMENTED** [review]
LGTM! Interesting stuff. Nice that you're vendoring htmx, this is good practise and we should do it for a lot more of our dependencies. I worked on something to help us do that yesterday, might add it this week.

File: app/tests/challenges_tests/test_forms.py (hunk 3 above)
Comment (amickan) [reply app/tests/challenges_tests/test_forms.py]: Neither are required here, thanks for catching this!

File: app/grandchallenge/challenges/views.py [hunk 4]
```diff
@@ -368,4 +371,24 @@ def form_valid(self, form):
             send_challenge_status_update_email(
                 challengerequest=form.instance, challenge=challenge
             )
+
+        response = HttpResponse()
+        response["HX-Refresh"] = "true"
+        return response
+
+
+class ChallengeRequestBudgetUpdate(
```
Comment (jmsmkn) [inline app/grandchallenge/challenges/views.py:380]: Sorry forgot this comment, but is there a test for this view? Both for permissions and functionality.

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**